### PR TITLE
Schedule cron job once a minute to send pending updates alternative

### PR DESCRIPTION
**Pull request by lezama**
alternative to #4471

### CODE REVIEW

File: sync/class.jetpack-sync-sender.php [hunk 1]
```diff
@@ -159,13 +170,25 @@ public function do_sync() {
...
+				$this->do_sync();
+			} else {
 				$this->schedule_sync( '+1 minute' );
 			}
 		}
 	}
 
 	private function schedule_sync( $when ) {
-		wp_schedule_single_event( strtotime( $when ), 'jetpack_sync_actions' );
+		if ( ! wp_next_scheduled( 'jetpack_sync_actions' ) ) {
```
Comment (gravityrail) [inline sync/class.jetpack-sync-sender.php]: I like this - we should do this part regardless of other stuff!

**Comment by gravityrail** [comment]
Some thoughts:

- I think scheduling cron to run a sync once per minute, rather than scheduling lots of individual cron events, is cleaner (as per #4471). This code adds a bunch of complexity to the do_sync method that makes it harder to understand how it behaves; the other PR removes that complexity.
- The other cron can be made to exit if do_sync returns false, indicating that the rate limit has been hit or there are no items in the queue (I have this change running, haven't pushed it yet).
- From a silly "purist" point of view, I like the idea of getting the scheduling code out of the sender - cleaner separation of concerns, as scheduling is an "integration" concern, rather than a "sending" concern :)
- One thing the other patch doesn't do is wait briefly in case the rate limit is about to expire - that would allow the rate to be higher. However, we could add that by adding a method to the sender that lets us ask when the rate limit is due to expire, and having cron `sleep()` until that time, provided it's < 10s or something.

**Comment by lezama** [comment]
Agreed that separating concerns is always better.

I agree that it adds complexity but it also simplifies in the way that `do_sync` always runs as a cron job in this version.

> The other cron can be made to exit if do_sync returns false, indicating that the rate limit has been hit or there are no items in the queue (I have this change running, haven't pushed it yet).

looking forward to that change!

let's go your way as you are the expert in the sync field, I was just trying to maybe give some inspiration :)

**Comment by gravityrail** [comment]
Closed in favour of #4471

**Pull request closed without merging**